Arnej/configurable legacy query parsing

arnej27959 opened the pull request:
@bratseth please review
@baldersheim please take a look also
@geirst @havardpe FYI

arnej27959 committed:
add feature flag "use-legacy-wand-query-parsing"

arnej27959 committed:
add some configuration for legacyQueryParsing

arnej27959 committed:
wire feature flag

arnej27959 committed:
modernize slightly

arnej27959 committed:
maybe fold things into wand

arnej27959 committed:
more careful defaults, more unit tests

bratseth reviewed (dismissed):
Look solid to me. 

One though: Once this is wired as throughly as this I think it will likely find other uses when we want to tweak parsing behavior/defaults and so be here to stay until Vespa is carried to the one million galaxies we'll eventually rule. So maybe it would be better to call it "ParserSettings" or something like that?

arnej27959 committed:
remove special handling of implicit phrases

* after discussion of PR, we probably will not need to do
  special handling of (implicit) phrase items, so remove
  that code - can easily be re-done later if needed.

arnej27959 committed:
rename LegacyQueryParsing -> ParserSettings
